### PR TITLE
Tests: Appveyor: don't fail build on upload failure.

### PR DESCRIPTION
**Pull request by xoviat**
The tests API is now failing for some reason, but that shouldn't fail the build.

See appveyor/ci#1363.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -173,4 +173,9 @@ on_success:
 
 
 on_finish:
-  - ps: 'if (Test-Path ".\junit-results.xml") { (new-object net.webclient).UploadFile("https://ci.appveyor.com/api/testresults/junit/$($env:APPVEYOR_JOB_ID)", (Resolve-Path .\junit-results.xml)) }'
+  - ps: |
+      (new-object net.webclient).UploadFile(
+        "https://ci.appveyor.com/api/testresults/junit/$($env:APPVEYOR_JOB_ID)",
+        (Resolve-Path .\junit-results.xml)
+      )
+      $LastExitCode = 0
```
Comment (htgoebel) [inline appveyor.yml]: Nice trick. Can you please add this for the cache-upload, too?

File: appveyor.yml [hunk 2]
```diff
@@ -173,6 +173,7 @@ on_success:
 
 
 on_finish:
+  - appveyor PushArtifact junit-results.xml
```
Comment (htgoebel) [inline appveyor.yml]: Please add some comment about what this is doing in difference to the UploadFile thing below.

But isn't this duplicate with the UploadFile stuff below?

**Comment by htgoebel** [comment]
@xoviat I'd ask you again to set up your own appveyor project and test your commits (on a different branch) there first. Our main build-queue is blocked with mini and minor changes and lagging a day. When using your own queue, you could chancel failing tests early and remove queue entries. This is what I'm doing a lot - esp. when testing CI behaviour. Thanks.

**Comment by htgoebel** [comment]
@xoviat Addendum: Also please open pull-requests for code you consider to be finished and passed your CI. Otherwise I'm spending a lot of time on code still in progress.

File: appveyor.yml (hunk 1 above)
Comment (xoviat) [inline appveyor.yml]: I haven't seen a cache upload failure fail the build. In fact, if you look [here](https://ci.appveyor.com/project/xoviat/pyinstaller/build/1.0.9/job/14i71eaqa4itto0i), you can see the cache failed but the build passed.

File: appveyor.yml (hunk 2 above)
Comment (xoviat) [inline appveyor.yml]: This not a  duplicate, but code that appveyor requested for debugging purposes. I will remove it.

**Comment by xoviat** [comment]
> @xoviat I'd ask you again to set up your own appveyor project and test your commits (on a different branch) there first. Our main build-queue is blocked with mini and minor changes and lagging a day. When using your own queue, you could chancel failing tests early and remove queue entries. This is what I'm doing a lot - esp. when testing CI behaviour. Thanks.

Well, you point out (rightly) issues with my pull requests and then I need to fix them. I try my best to get things right the first time, but that hasn't been very successful (obviously).

**Comment by xoviat** [comment]
Is there anything more to be done on this?

**Comment by xoviat** [comment]
Appveyor is now fixed, bit this should still be merged so that the builds won't fail if there is a problem with junit XML in the future.

**Comment by htgoebel** [comment]
Thanks for keeping an eye on it. So we keep  `UploadFile` and are not going to switch to `PushArtifact`? Totally fine for me. I'll merge this the next days.

**Comment by xoviat** [comment]
> Thanks for keeping an eye on it. So we keep UploadFile and are not going to switch to PushArtifact? Totally fine for me. I'll merge this the next days.

The PushArtifact is not a replacement for UploadFile. It was executed in addition to UploadFile on some builds to provide appveyor support with additional debugging information. I removed it after they no longer required it.

**Comment by htgoebel** [comment]
This as later merges in 22b868ae93eadd4034419824ebcae53f3c1712fc

**Pull request closed without merging**